eng/Generation.psm1

```diff
@@ -50,6 +50,59 @@ function AutoRest-Reset()
     Invoke "$script:autoRestBinary --reset"
 }
 
+function Invoke-Cadl($baseOutput, $projectName, $sharedSource="", $fast="", $debug="")
+{
+    $baseOutput = Resolve-Path -Path $baseOutput
+    $baseOutput = $baseOutput -replace "\\", "/"
+    $outputPath = Join-Path $baseOutput "Generated"
+    $outputPath = $outputPath -replace "\\", "/"
+    #clean up
+    if (Test-Path $outputPath) {
+        Remove-Item $outputPath/* -Include *.* -Exclude *Configuration.json*
+    }
+    # emit cadl json
+    $repoRootPath = Join-Path $PSScriptRoot ".."
+    $repoRootPath = Resolve-Path -Path $repoRootPath
+    Push-Location $repoRootPath
+    # node node_modules\@cadl-lang\compiler\dist\core\cli.js compile --output-path $outputPath "$baseOutput\$projectName.cadl" --emit @azure-tools/cadl-csharp
+    $emitCommand = "node node_modules/@cadl-lang/compiler/dist/core/cli.js compile --output-path $outputPath $baseOutput/$projectName.cadl --emit @azure-tools/cadl-csharp"
+    Invoke $emitCommand
+    Pop-Location
+    
+    $dotnetArguments = $debug ? "--no-build --debug" : "--no-build" 
+    $command = "dotnet run --project $script:AutoRestPluginProject $dotnetArguments --standalone $outputPath"
+    Invoke $command
+    Invoke "dotnet build $baseOutput --verbosity quiet /nologo"
+}
+
+function Invoke-CadlSetup()
+{
+    # build emitter
+    $emitterPath = Join-Path $PSScriptRoot ".." "src" "CADL.Extension" "Emitter.Csharp"
+    $emitterPath = Resolve-Path -Path $emitterPath
+    Push-Location $emitterPath
+    npm install
```

Comment:
I think we should run `npm ci`

Reply:
updated

Reply:
Sorry, seems not changed?

Reply:
I think it's changed to `npm ci`. You probably need to refresh the page.